scalajvm/app/security/BasicAuth.scala

```diff
@@ -59,5 +56,9 @@ case class BasicAuth[A](action: Action[A]) extends Action[A] {
     }
   }
 
+  def executionContext = scala.concurrent.ExecutionContext.global
+
   lazy val parser = action.parser
-}
\ No newline at end of file
+
+  private final val decoder = Base64.getDecoder()
```

Comment:
I don't think we need this `final` here. It stops child classes from overriding the field, but we mostly don't care (in this project, at least).

`val` is already immutable enough, which is probably what you were looking for.

Reply:
I don't think subclasses can override `private` fields, are you sure? And I think subclasses will want their own instance anyway: there are a couple decoder types all having the same type, and subclasses probably want to make sure they're using the kind *they* need, not the one that some base class uses.

I can change this if you insist, but I honestly don't see a reason to.

Reply:
No, the subclasses aren't allowed to override `private` properties. And that's the reason this `final` is absolutely unnecessary here.

[Here's an explanation](https://stackoverflow.com/a/13412894/2684760) what `private final val` does. I don't think it's relevant for our case, so I'd still insist on removing it here.

Reply:
Ah, okay, so `final` here is just an unnecessary optimization. Thanks for the link! I removed the modifier in https://github.com/codingteam/loglist/pull/225/commits/9f3d95f34315160fd8fc60e346c32e776ee35214.